p2p/src/message.rs

```diff
@@ -21,48 +21,43 @@ use common::{
 use serialization::{Decode, Encode};
 
 #[derive(Debug, Encode, Decode, Clone, PartialEq, Eq)]
-pub enum SyncingRequest {
-    #[codec(index = 0)]
-    GetHeaders { locator: Vec<BlockHeader> },
-    #[codec(index = 1)]
-    GetBlocks { block_ids: Vec<Id<Block>> },
+pub struct HeaderRequest {
```

Comment:
Why aren't these types encapsulated with proper factor functions and private field(s)? This looks more like a C struct.

Reply:
Fixed